src/time/tests/WaveformTest.cpp

```diff
@@ -0,0 +1,214 @@
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 0), 2.0));
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 2.0));
+
+  value(0) = 3.0;
+  waveform.storeAtFirstSample(value);
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 3.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 3.0));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 3.0));
+}
+
+BOOST_AUTO_TEST_CASE(testInterpolateDataFirstOrder)
+{
+  PRECICE_TEST(1_rank);
+
+  testing::WaveformFixture fixture;
+
+  // Test first order interpolation
+  const int interpolationOrder = 1;
+  Waveform  waveform(interpolationOrder);
+  const int valuesSize = 1;
+  waveform.initialize(valuesSize);
+
+  BOOST_TEST(fixture.maxNumberOfStoredSamples(waveform) == 2);
+  BOOST_TEST(fixture.valuesSize(waveform) == 1);
+  BOOST_TEST(fixture.numberOfStoredSamples(waveform) == 1);
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 0), 0.0));
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 1), 0.0));
+
+  Eigen::VectorXd value(1);
+  value(0) = 1.0;
+  waveform.storeAtFirstSample(value);
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 0), 1.0));
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 1), 0.0));
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 1.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 1.0));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 1.0));
+
+  value(0) = 2.0;
+  waveform.storeAtFirstSample(value);
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 2.0));
+
+  waveform.moveToNextWindow();
+  BOOST_TEST(fixture.numberOfStoredSamples(waveform) == 2);
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 0), 2.0));
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 1), 2.0));
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 2.0));
+
+  value(0) = 3.0;
+  waveform.storeAtFirstSample(value);
+
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 0), 3.0));
+  BOOST_TEST(testing::equals(fixture.getValue(waveform, 0, 1), 2.0));
+
+  BOOST_TEST(testing::equals(waveform.sample(0.0)(0, 0), 2.0));
+  BOOST_TEST(testing::equals(waveform.sample(0.5)(0, 0), 2.5));
+  BOOST_TEST(testing::equals(waveform.sample(1.0)(0, 0), 3.0));
+}
+
+BOOST_AUTO_TEST_CASE(testInterpolateDataSecondOrder)
+{
+  PRECICE_TEST(1_rank);
+
+  testing::WaveformFixture fixture;
+
+  // Test second order interpolation
+  const int interpolationOrder = 2;
+  Waveform  waveform(interpolationOrder);
+  const int valuesSize = 1;
```

Comment:
All your current tests have a value size of 1. For test coverage it would be good to have a value > 1 here as well.

Reply:
I added a simple test in 0f28da8b118865aadbf81f715db3c2f2043b369f.